src/App.js

```diff
@@ -1,16 +1,75 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog.js';
+import { useState } from 'react';
+import ColorChoice from './components/ColorChoice';
+
 
 const App = () => {
+  const [chatData, setChatData] = useState(chatMessages)
+
+  // Process message entries when like button is selected
+  const fillHeart = (id) => {
+    setChatData(chatData => chatData.map(entry => {
+      if (entry.id === id) {
+        return {...entry, liked: !entry.liked};
+      } else {
+        return entry;
+      }
+    }))
+  };
+
+  // Count liked messages and tally total
+  const countFilledHearts = (chatData) => {
+    return chatData.reduce((total, entry) => {
+      return entry.liked ? total + 1 : total;
+    }, 0)
+  };
+
+  const totalLikedMessages = countFilledHearts(chatData);
+
+  // Determine local and remote sender
+  const localSender = chatMessages[0]['sender'];
+  let remoteSender = '';
+  for (const entry of chatMessages) {
+    if (entry.sender !== localSender) {
+      remoteSender = entry.sender;
+    }
+  };
+
+  // Establish default sender colors
+  const [color, setColor] = useState({ local: 'blue', remote: 'green'});
```

Comment:
It's React best practice to group the state variables at the top of the component so it's easy to find -- move this below where the `chatData` state is declared!